### ISSUE TITLE
How come it work with two different models?

### ISSUE THREAD

**Issue by LuciAkirami** [timinar/BabyLlama#6]
To perform a knowledge distillation, the teacher and the student model must have the same tokenizer. But you have used two different teacher models, that is Llama and GPT2, and these both have different tokenizers, thus their probabilities cannot be compared. So my doubt is, how come it works?

**Comment by timinar** [comment]
We use one and the same tokenizer trained on the BabyLM training dataset for all models. Llama and GPT2 refer to the architectures in HF Transformers, not the original pretrained models.

**Comment by LuciAkirami** [comment]
So you have trained a custom Llama and GPT2 models with your custom tokenizer. And then you have used the same custom tokenizer to train the BabyLlama as well. Am I right?

**Comment by timinar** [comment]
Yes, exactly.

**Comment by LuciAkirami** [comment]
Oh thank you. It's mentioned in the paper, and I just read it. So you have trained it on an RTX 3090 right? What is the peak memory usage?

**Comment by timinar** [comment]
Yes. If I remember correctly, it was  below 20 GB with the default hyper parameters (still ~100% utilization). You can always reduce it by decreasing the batch size and increasing `gradient_accumulation_steps`.

**Comment by LuciAkirami** [comment]
Thanks for the info. I'm planning on running it on a 3060 with 12GB VRAM, I think I have to drastically reduce the batch size and increase `gradient_accumulation_steps` by a bit

**Comment by timinar** [comment]
Well, if you don't want to change the lr,  proportionally increase `gradient_accumulation_steps ` so that the logical_bsz =  gradient_accumulation_steps * bsz stays the same.

**Comment by LuciAkirami** [comment]
logical batch size? Sorry I did not get you

**Comment by timinar** [comment]
Ah, I mean that with the gradient accumulation you have effective (or logical) batch size which is equal to bsz * gradient_accumulation_steps
You want to keep this effective batch size constant.

**Comment by LuciAkirami** [comment]
So I need to try to match my logical bsz to your actual bsz by increasing the gradient accumulation

**Comment by timinar** [comment]
Yep
batch_size -> batch_size / factor
gradient_accumulation_steps ->  gradient_accumulation_steps * factor

I would start with factor = 2, and increase till it fits.

**Comment by LuciAkirami** [comment]
Once again, thanks a lot for the clarification. Closing this issue by the way
